File: 523. Continuous Subarray Sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        prefix_map = {}
        prefix_map[0] = -1
        current = 0
        for i in range(len(nums)):
            current += nums[i]
            current %= k
            if (current+k)%k in prefix_map and i-prefix_map[(current+k)%k]>=2:
                return True
            if current not in prefix_map:
                prefix_map[current] = i
        return False

File: 523. Continuous Subarray Sum.go
func checkSubarraySum(nums []int, k int) bool {
    prefixMap := make(map[int]int)
    prefixMap[0] = -1
    current := 0
    for i := 0; i < len(nums); i++ {
        current += nums[i];
        current %= k
        if index, ok := prefixMap[(current + k) % k]; ok && i-index >= 2 {
            return true
        }
        if _, ok := prefixMap[current]; !ok {
            prefixMap[current] = i
        }
    }
    return false
}
